test/widgets/compose_box_test.dart

```diff
@@ -49,6 +56,11 @@ void main() {
     await store.addUsers([eg.selfUser, ...users]);
     connection = store.connection as FakeApiConnection;
 
+    if (textScaleFactor != null) {
+      tester.platformDispatcher.textScaleFactorTestValue = textScaleFactor;
+      addTearDown(tester.platformDispatcher.clearTextScaleFactorTestValue);
```

Comment:
Can this step just go in the test cases that need it? That way we avoid making this shared helper more complex with a growing number of optional features, plus it makes it transparent in the test cases exactly what's happening.

If this step were a bit more code than this (or if we were writing lots of very short tests that used it), another tactic would be to make a helper function just for this, which those tests could call.

Reply:
Inlined these without a helper, because they are short enough.